packages/api/src/dataProviders/api/rechercheEntreprises/rechercheEntreprises.port.ts

```diff
@@ -16,17 +16,35 @@ export class RechercheEntreprises {
     }
 
     async search(query: string) {
-        const answer = await this.http.get<RechercheEntreprisesDto>(
-            RechercheEntreprises.URL + `?q=${query}&nature_juridique=${RechercheEntreprises.natureJuridique}`,
-        );
+        const results: AssociationNameEntity[] = [];
+        let page = 1;
+        let response = (
+            await this.http.get<RechercheEntreprisesDto>(
+                RechercheEntreprises.URL +
+                    `?q=${query}&nature_juridique=${RechercheEntreprises.natureJuridique}&per_page=25`,
+            )
+        ).data;
 
-        return (
-            answer.data.results?.reduce((acc, result) => {
-                if (!result.nom_complet || !result.siren) return acc;
+        const addResults = (response: RechercheEntreprisesDto) =>
+            response.results?.map(result => {
+                if (!result.nom_complet || !result.siren) return;
                 const dto = result as RechercheEntreprisesResultDto & { siren: string; nom_complet: string }; // tell ts that the new typing is good
-                return acc.concat([RechercheEntreprisesAdapter.toAssociationNameEntity(dto)]);
-            }, [] as AssociationNameEntity[]) || []
-        );
+                return results.push(RechercheEntreprisesAdapter.toAssociationNameEntity(dto));
+            });
+
+        const maxPage = response.total_pages;
+        for (page = 2; page <= Math.min(maxPage, 3); page++) {
+            addResults(response);
```

Comment:
Pourquoi deux fois addResults dans la boucle `for` ? au second call on va ajouter deux fois le même résultat, non ?

Reply:
Yen a qu'un seul dans la boucle for, le second se fait après la boucle, et correspond au dernier appel

Reply:
autant pour moi je trouve pas très lisible la relecture GitHub sur un petit écran

Reply:
Ah si je me suis ratée. Je reprends ça